src/guides/optimism-bridge-widget/index.md

```diff
@@ -0,0 +1,1877 @@
+    uint nftsIndex = 0;
+    for (uint i = 0; i < nftCount; i++) {
+      if (_idToNFT[i + 1].owner == msg.sender) {
+        nfts[nftsIndex] = _idToNFT[i + 1];
+        nftsIndex++;
+      }
+    }
+    return nfts;
+  }
+
+  function getMyListedNfts() public view returns (NFT[] memory) {
+    uint nftCount = _nftCount.current();
+    uint myListedNftCount = 0;
+    for (uint i = 0; i < nftCount; i++) {
+      if (_idToNFT[i + 1].seller == msg.sender && _idToNFT[i + 1].listed) {
+        myListedNftCount++;
+      }
+    }
+
+    NFT[] memory nfts = new NFT[](myListedNftCount);
+    uint nftsIndex = 0;
+    for (uint i = 0; i < nftCount; i++) {
+      if (_idToNFT[i + 1].seller == msg.sender && _idToNFT[i + 1].listed) {
+        nfts[nftsIndex] = _idToNFT[i + 1];
+        nftsIndex++;
+      }
+    }
+    return nfts;
+  }
+}
+
+```
+
+</p>
+</details>
+
+### Modify `truffle-config.ovm.js`
+
+Now that we want to reference our new Optimsim contracts in the client, we need to change the build path in `truffle-config.ovm.js` to add the contract ABIs into a folder in `client`. Modify the build directory in `truffle-config.ovm.js` as follows:
+
+```javascript
+contracts_build_directory: "../client/contracts/optimism-contracts",
+```
+
+### Migrate the new Optimism contracts
+
+Now, let's deploy the newly created Optimism contracts. To do so, create a migrations file `1_deploy_contracts.js`. Then populate as so:
+
+```javascript
+var OptimismNFT = artifacts.require("OptimismNFT");
+var OptimismMarketplace = artifacts.require("OptimismMarketplace");
+
+module.exports = async function(deployer) {
+  await deployer.deploy(OptimismMarketplace);
+  const marketplace = await OptimismMarketplace.deployed();
+  await deployer.deploy(OptimismNFT, marketplace.address);
+}
+```
+
+Since we are only deploying these two contracts, which are both on Optimism, let's deploy using Truffle Dashboard! First, let's add the Optimism Goerli network to our MetaMask wallet. You should add this network information to your MetaMask wallet:
+
+- **Network name**: Optimism Goerli
+- **New RPC URL**: https://optimism-goerli.infura.io/v3/<INFURA_API_KEY>
+- **Chain ID**: 420
+- **Currency Symbol**: ETH
+- **Block explorer URL**: https://blockscout.com/optimism/goerli/
+
+Now, to migrate your contracts first start up dashboard on `localhost:24012`:
+
+```shell
+truffle dashboard
+```
+
+Make sure the Optimism Goerli network is selected on MetaMask. Then simply call:
+
+```shell
+truffle migrate --config truffle-config.ovm.js --network dashboard
+```
+
+Truffle dashboard will ask you sign to transactions:  one for deploying `OptimismMarketplace` and one for deploying `OptimismNFT`. You should be able to see the built contracts in `client/contract/optimism-contracts`.
+
+At this point, you should test the contracts. We won't be going over it in this tutorial, but you can follow the instructions [here](https://trufflesuite.com/guides/nft-marketplace/#test-your-smart-contracts)
```

Comment:
Need a full stop at the end of the following sentence.

Also noticing a lot of URLs are absolute (e.g. including the full https://trufflesuite.com domain)...probably should be relative.